Feat: Member 기능 구현

yugyeom-ghim opened the pull request:
## 구현 한 내용
- 로그인에 대한 테스트는 시간이 부족해서 아직 작성을 못했습니다. 시간이 되면 작성할거같고 정 안되면 안드로이드분들 카카오 로그인 로직 구현 다 됐을때 같이 테스트 해도 좋을 거 같습니다!
- AccessToken을 클라이언트에서 준다고 하셔서 안드로이드에서 이미 검증 된 Kakao AccessToken을 받는다 라고 생각하여 작성하였습니다.
- Interceptor에서 특정 path를 제외하고 토큰이 유효한지 검증했습니다.
- @Auth 어노테이션이 붙어있으면서 Long타입인 파라미터인 경우 ArgumentResolver에서 토큰검증 및 memberId를 반환해주도록 구현했습니다.
- findMyProfile 메서드는 저희가 정한 명명규칙이랑 맞지않긴한데,, 이정도 유연성은 허용해야한다고 해서 이렇게 작성했습니다.


## 완료 한 이슈
Closes #11 
Closes #12
Closes #13

yugyeom-ghim committed:
Feat: Member 도메인 구현

yugyeom-ghim committed:
Feat: Kakao 로그인 Client로직 구현

- 안드로이드에서 전달 된 이미 검증이 된 Kakao accesstoken으로 Member조회

yugyeom-ghim committed:
Test: Kakao 로그인 fetchMember 단위테스트 작성

yugyeom-ghim committed:
Feat: JWT 관련 기능 구현

- Interceptor에서 특정 path를 제외하고 토큰이 유효한지 검증
- @Auth 어노테이션이 붙어있으면서 Long타입인 파라미터인 경우 ArgumentResolver에서 토큰검증 및 memberId를 반환해주도록 구현

yugyeom-ghim committed:
Feat: Member API 구현

hynseoj commented on src/test/java/notai/client/oauth/kakao/KakaoOauthClientTest.java:
```diff
@@ -0,0 +1,50 @@
+                12345L,
+                true,
+                null,
+                new KakaoMemberResponse.KakaoAccount(
```
이 부분 변수로 빼면 더 좋을 것 같아요~!

yugyeom-ghim replied:
반영했습니다!

hynseoj reviewed (approved):
고생하셨습니다!!!

yugyeom-ghim committed:
Style: 불필요한 import 제거

yugyeom-ghim committed:
Test: 하드코딩 된 값 변수추출

yugyeom-ghim committed:
Refactor: 불필요한 토큰 검증 제거

- Interceptor에서 검증된 토큰으로 memberId를 추출해 ArgumenResolver에서 그 memberId를 사용할 수 있도록 개선

Shsin9797 commented:
굿!

yunjunghun0116 reviewed (approved):
고생하셨습니다 :)
